lib/driver.rb

```diff
@@ -0,0 +1,44 @@
+require 'csv'
+
+module RideShare
+    class Driver
+        DRIVER_INFO = CSV.read('support/drivers.csv')
+
+        attr_accessor :driver_id, :name, :vin
+
+        def initialize(id)
+            raise ArgumentError, "#{id} is not a valid ID number, only positive whole numbers are accepted" unless id.is_a?(Integer) && id >= 1
+            raise ArgumentError, "#{id} Cannot be found" unless (DRIVER_INFO.map { |line| line[0].to_i }).include? id
+
+            DRIVER_INFO.each do |line|
+                next unless line[0].to_i == id
+                @driver_id = id
```

Comment:
I like the idea here - being able to initialize a driver with just the ID and have the program fill in the details is very convenient. However, I'm a little worried about the efficiency of looping through the whole `DRIVER_INFO` array here. In particular, if `Driver.all` creates a new `Driver` for every ID in `DRIVER_INFO`, then `Driver.all` ends up being `O(n^2)`.

It might be you could chose a different data structure to get around this. I'll leave that up to you.